src/main/cpp/auton/CyclePrimitives.cpp

```diff
@@ -53,7 +53,7 @@ CyclePrimitives::CyclePrimitives() : State(string("CyclePrimitives"), 0),
 									 m_currentPrim(nullptr),
 									 m_primFactory(PrimitiveFactory::GetInstance()),
 									 m_driveStop(nullptr),
-									 m_autonSelector(new AutonSelector()),
+									 m_autonSelector(AutonSelector::GetInstance()),
```

Comment:
It might be cleaner to not make autonselector a singleton.  Rather, change the signature of this method to take in the 3 delay times.  Then after you get the delays back, you can loop over them looking for DriveStopDelay types and then calling a method to set the delay time explicitly.